tests/integration/Core/Framework/MessageQueue/Stats/MySQLStatsRepositoryTest.php

```diff
@@ -0,0 +1,119 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Integration\Core\Framework\MessageQueue\Stats;
+
+use Doctrine\DBAL\Connection;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\MessageQueue\Stats\MySQLStatsRepository;
+use Shopware\Core\Framework\Test\TestCaseBase\KernelTestBehaviour;
+
+/**
+ * @internal
+ */
+#[Package('framework')]
+#[CoversClass(MySQLStatsRepository::class)]
+class MySQLStatsRepositoryTest extends TestCase
+{
+    use KernelTestBehaviour;
+
+    private Connection $connection;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->connection = $this->getContainer()->get('Doctrine\DBAL\Connection');
+        $this->cleanTable();
+    }
+
+    protected function tearDown(): void
+    {
+        $this->cleanTable();
+    }
+
+    public function testUpdateMessageStatsDeletesExpired(): void
+    {
+        $repository = new MySQLStatsRepositoryTestable($this->connection, 20);
+
+        $now = 1726671956;
+        $repository->setNow($this->dateTimeFromTime($now - 21));
```

Comment:
instead of magic numbers, using some constants would make the purpose more clearer. 

e.g. `TOO_OLD` or something like that